main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripTravelSpeedValidator.java

```diff
@@ -0,0 +1,131 @@
+  @Inject GtfsTripTableContainer tripTable;
+  @Inject GtfsStopTimeTableContainer stopTimeTable;
+  @Inject GtfsStopTableContainer stopTable;
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    for (Entry<String, List<GtfsStopTime>> entry :
+        Multimaps.asMap(stopTimeTable.byTripIdMap()).entrySet()) {
+      List<FastTravelBetweenStopsNotice> noticesForTrip = checkSpeedAlongTrip(entry);
+      for (FastTravelBetweenStopsNotice notice : noticesForTrip) {
+        noticeContainer.addValidationNotice(notice);
+      }
+    }
+  }
+
+  /**
+   * Calculates the travel speed between {@code GtfsStopTime}. Generates a {@code
+   * FastTravelBetweenStopsNotice} if the travel speed between {@code GtfsStopTimes} is greater than
+   * 150 km/h.
+   *
+   * @param tripStopTimes the list of {@code GtfsStopTimes} sorted by stop_times.stop_sequence for a
+   *     {@code GtfsTrip}
+   * @return the list of {@code FastTravelBetweenStopsNotice} generated when checking speed travel
+   *     along the given trip
+   */
+  private List<FastTravelBetweenStopsNotice> checkSpeedAlongTrip(
+      Entry<String, List<GtfsStopTime>> tripStopTimes) {
+    var previousStopsData =
+        new Object() {
+          final List<Integer> accumulatedStopSequence = new ArrayList<>();
+          GtfsTime departureTime = null;
+          double latitude;
+          double longitude;
+          // used to accumulate distance between stops with same arrival and departure
+          // times
+          int accumulatedDistanceMeter = 0;
+        };
+    List<FastTravelBetweenStopsNotice> notices = new ArrayList<>();
+    for (GtfsStopTime stopTime : tripStopTimes.getValue()) { // prepare data for current iteration
+      GtfsStop currentStop = stopTable.byStopId(stopTime.stopId());
+      double currentStopLat = currentStop.stopLat();
+      double currentStopLon = currentStop.stopLon();
+      GtfsTime currentArrivalTime = stopTime.arrivalTime();
+      double distanceFromPreviousStopMeter =
+          GeospatialUtil.distanceInMeterBetween(
+              previousStopsData.latitude,
+              previousStopsData.longitude,
+              currentStopLat,
+              currentStopLon);
+      boolean sameArrivalAndDeparture = false;
+      if (previousStopsData.departureTime != null && currentArrivalTime != null) {
+        sameArrivalAndDeparture = currentArrivalTime.equals(previousStopsData.departureTime);
+        if (!sameArrivalAndDeparture) {
+          int durationSecond =
+              currentArrivalTime.getSecondsSinceMidnight()
+                  - previousStopsData.departureTime.getSecondsSinceMidnight();
+          double distanceMeter =
+              distanceFromPreviousStopMeter + previousStopsData.accumulatedDistanceMeter;
+          double speedMeterPerSecond = distanceMeter / durationSecond;
+          if (speedMeterPerSecond > 42) { // roughly 150 km per hour. Put it in default parameters
```

Comment:
`42` should be pulled out into it's own static final constant at the top of this class. I'd make it a `double` too, and add in a comment what the value is in mph as well.